internal/server/server.go

```diff
@@ -75,9 +104,70 @@ func New(opts ...ServerFuncOpt) (http.Handler, error) {
 	return s, nil
 }
 
+func defaultExternalFuncs() *ExternalFuncs {
+	return &ExternalFuncs{
+		oidcNewProvider:           oidc.NewProvider,
+		oauth2ConfigExchange:      (*oauth2.Config).Exchange,
+		oidcIDTokenVerifierVerify: (*oidc.IDTokenVerifier).Verify,
+		oidcIDTokenClaims:         (*oidc.IDToken).Claims,
+	}
+}
+
+func httpClientForRootCAs(rootCABytes []byte) (*http.Client, error) {
+	tlsConfig := tls.Config{RootCAs: x509.NewCertPool()}
+	if !tlsConfig.RootCAs.AppendCertsFromPEM(rootCABytes) {
+		return nil, fmt.Errorf("no certs found in rootCABase64")
+	}
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tlsConfig,
+			Proxy:           http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		},
+	}, nil
+}
+
+func (s *Server) ConfigureOpenID() error {
+	var err error
+
+	if s.clusterCA != "" {
+		// decode root certificates
+		rootCABytes, err := base64.StdEncoding.DecodeString(s.clusterCA)
+		if err != nil {
+			return err
+		}
+		// get HTTP Client with custom root CAs
+		s.client, err = httpClientForRootCAs(rootCABytes)
+		if err != nil {
+			return err
+		}
+	} else {
+		s.client = &http.Client{
+			Timeout: 10 * time.Second,
+		}
+	}
+
+	s.context = oidc.ClientContext(context.Background(), s.client)
+	s.provider, err = s.externalFuncs.oidcNewProvider(s.context, s.issuerURL.String())
```

Comment:
No need to store this in the struct, you can just pass it directly to the provider.
```suggestion
	ctx := oidc.ClientContext(context.Background(), s.client)
	s.provider, err = s.externalFuncs.oidcNewProvider(ctx, s.issuerURL.String())
```

Reply:
Fixed